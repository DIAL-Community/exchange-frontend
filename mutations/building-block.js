import { gql } from '@apollo/client'

export const CREATE_BUILDING_BLOCK = gql`
  mutation CreateBuildingBlock(
    $name: String!
    $slug: String!
    $maturity: String!
    $imageFile: Upload
    $description: String!
    $specUrl: String
  ) {
    createBuildingBlock(
      name: $name
      slug: $slug
      maturity: $maturity
      imageFile: $imageFile
      description: $description
      specUrl: $specUrl
    ) {
      buildingBlock {
        slug
      }
      errors
    }
  }
`
