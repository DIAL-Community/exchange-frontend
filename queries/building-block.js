import { gql } from '@apollo/client'

export const BUILDING_BLOCK_DETAIL_QUERY = gql`
  query BuildingBlock($slug: String!) {
    buildingBlock(slug: $slug) {
      id
      name
      slug
      imageFile
      category
      specUrl
      buildingBlockDescription {
        description
        locale
      }
      workflows {
        name
        slug
        imageFile
      }
      products {
        name
        slug
        imageFile
        buildingBlocksMappingStatus
      }
    }
  }
`

export const BUILDING_BLOCK_SEARCH_QUERY = gql`
  query BuildingBlocks($search: String!) {
    buildingBlocks(search: $search) {
      id
      name
      slug
      imageFile
      maturity
    }
  }
`

export const BUILDING_BLOCK_QUERY = gql`
  query BuildingBlock($slug: String!) {
    buildingBlock(slug: $slug) {
      id
      name
      slug
      specUrl
      maturity
      category
      buildingBlockDescription {
        description
        locale
      }
    }
  }
`

export const BUILDING_BLOCKS_QUERY = gql`
  query SearchBuildingBlocks(
    $first: Int,
    $after: String,
    $sdgs: [String!],
    $useCases: [String!],
    $workflows: [String!],
    $categoryTypes: [String!],
    $showMature: Boolean,
    $search: String!
    ) {
    searchBuildingBlocks(
      first: $first,
      after: $after,
      sdgs: $sdgs,
      useCases: $useCases,
      workflows: $workflows,
      categoryTypes: $categoryTypes,
      showMature: $showMature,
      search: $search
    ) {
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasPreviousPage
        hasNextPage
      }
      nodes {
        id
        name
        slug
        imageFile
        maturity
        category
        specUrl
        workflows {
          slug
          name
          imageFile
        }
        products {
          slug
          name
          imageFile
        }
      }
    }
  }
`
