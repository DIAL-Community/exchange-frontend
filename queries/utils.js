const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
