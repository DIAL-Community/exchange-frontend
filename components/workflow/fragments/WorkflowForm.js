import React, { useCallback, useContext, useState } from 'react'
import { useRouter } from 'next/router'
import { Controller, useForm } from 'react-hook-form'
import { FaSpinner } from 'react-icons/fa6'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../lib/apolloClient'
import { ToastContext } from '../../../lib/ToastContext'
import FileUploader from '../../shared/form/FileUploader'
import { HtmlEditor } from '../../shared/form/HtmlEditor'
import Input from '../../shared/form/Input'
import ValidationError from '../../shared/form/ValidationError'
import { CREATE_WORKFLOW } from '../../shared/mutation/workflow'
import { PAGINATED_WORKFLOWS_QUERY, WORKFLOW_PAGINATION_ATTRIBUTES_QUERY } from '../../shared/query/workflow'
import { DEFAULT_PAGE_SIZE } from '../../utils/constants'

const WorkflowForm = React.memo(({ workflow }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const slug = workflow?.slug ?? ''

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const [updateWorkflow, { reset }] = useMutation(CREATE_WORKFLOW, {
    refetchQueries: [{
      query: WORKFLOW_PAGINATION_ATTRIBUTES_QUERY,
      variables: { search: '' },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }, {
      query: PAGINATED_WORKFLOWS_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }],
    onCompleted: (data) => {
      if (data.createWorkflow.workflow && data.createWorkflow.errors.length === 0) {
        const redirectPath = `/${locale}/workflows/${data.createWorkflow.workflow.slug}`
        const redirectHandler = () => router.push(redirectPath)
        setMutating(false)
        showSuccessMessage(
          format('toast.submit.success', { entity: format('ui.workflow.label') }),
          redirectHandler
        )
      } else {
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.workflow.label') }))
        setMutating(false)
        reset()
      }
    },
    onError: () => {
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.workflow.label') }))
      setMutating(false)
      reset()
    }
  })

  const { handleSubmit, register, control, formState: { errors } } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: workflow?.name,
      description: workflow?.workflowDescription?.description
    }
  })

  const doUpsert = async (data) => {
    setMutating(true)
    const { name, imageFile, description } = data
    const variables = {
      name,
      slug,
      description
    }

    if (imageFile) {
      variables.imageFile = imageFile[0]
    }

    updateWorkflow({
      variables,
      context: {
        headers: {
          'Accept-Language': locale
        }
      }
    })
  }

  const cancelForm = () => {
    setReverting(true)
    router.push(`/${locale}/workflows/${slug}`)
  }

  return (
    <form onSubmit={handleSubmit(doUpsert)}>
      <div className='px-4 lg:px-0 py-4 lg:py-6 text-dial-plum'>
        <div className='flex flex-col gap-y-6 text-sm'>
          <div className='text-xl font-semibold'>
            {workflow
              ? format('app.editEntity', { entity: workflow.name })
              : `${format('app.createNew')} ${format('ui.workflow.label')}`}
          </div>
          <div className='flex flex-col gap-y-2'>
            <label className='text-dial-sapphire required-field' htmlFor='name'>
              {format('workflow.name')}
            </label>
            <Input
              {...register('name', { required: format('validation.required') })}
              id='name'
              placeholder={format('workflow.name')}
              isInvalid={errors.name}
            />
            {errors.name && <ValidationError value={errors.name?.message} />}
          </div>
          <div className='flex flex-col gap-y-2'>
            <label className='text-dial-sapphire'>{format('workflow.imageFile')}</label>
            <FileUploader {...register('imageFile')} />
          </div>
          <div className='flex flex-col gap-y-2'>
            <label id='description' className='text-dial-sapphire required-field'>
              {format('workflow.description')}
            </label>
            <Controller
              name='description'
              control={control}
              render={({ field: { value, onChange } }) => (
                <HtmlEditor
                  labelledBy='description'
                  onChange={onChange}
                  initialContent={value}
                  placeholder={format('workflow.description')}
                />
              )}
              rules={{ required: format('validation.required') }}
            />
            {errors.description && <ValidationError value={errors.description?.message} />}
          </div>
          <div className='flex flex-wrap text-base mt-6 gap-3'>
            <button type='submit' className='submit-button' disabled={mutating || reverting}>
              {`${format('app.submit')} ${format('ui.workflow.label')}`}
              {mutating && <FaSpinner className='spinner ml-3' />}
            </button>
            <button
              type='button'
              className='cancel-button'
              disabled={mutating || reverting}
              onClick={cancelForm}
            >
              {format('app.cancel')}
              {reverting && <FaSpinner className='spinner ml-3' />}
            </button>
          </div>
        </div>
      </div>
    </form>
  )
})

WorkflowForm.displayName = 'WorkflowForm'

export default WorkflowForm
