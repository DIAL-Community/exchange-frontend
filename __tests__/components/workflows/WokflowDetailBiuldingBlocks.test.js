import { fireEvent, screen } from '@testing-library/react'
import { render, waitForAllEffectsAndSelectToLoad } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { BUILDING_BLOCK_SEARCH_QUERY } from '../../../queries/building-block'
import WorkflowDetailBuildingBlocks from '../../../components/workflows/WorkflowDetailBuildingBlocks'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import { buildingBlocks } from './data/WorkflowDetailBiuldingBlocks'
import { workflow } from './data/WorkflowForm'

mockNextUseRouter()
describe('Unit tests for the WorkflowDetailBuildingBlocks component.', () => {
  const EDIT_BUTTON_TEST_ID = 'edit-button'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const BUILDING_BLOCK_SEARCH_TEST_ID = 'building-block-search'
  const BUILDING_BLOCK_SEARCH_OPTION_1_LABEL = 'Building Block 1'
  const BUILDING_BLOCK_SEARCH_OPTION_2_LABEL = 'Building Block 2'
  const WORKFLOW_TEST_BUILDING_BLOCK_LABEL = 'Test Building Block'
  const PILL_TEST_ID = 'pill'
  const PILL_REMOVE_BUTTON_TEST_ID = 'remove-button'
  const mockBuildingBlocks = generateMockApolloData(BUILDING_BLOCK_SEARCH_QUERY, { search: '' }, null, buildingBlocks)

  describe('Should match snapshot -', () => {
    test('without edit permission.', () => {
      const { container } = render(
        <CustomMockedProvider mocks={[mockBuildingBlocks]}>
          <WorkflowDetailBuildingBlocks
            canEdit={false}
            workflow={workflow}
          />
        </CustomMockedProvider>
      )
      expect(container).toMatchSnapshot()
    })

    test('with edit permission.', () => {
      const { container } = render(
        <CustomMockedProvider mocks={[mockBuildingBlocks]}>
          <WorkflowDetailBuildingBlocks
            canEdit={true}
            workflow={workflow}
          />
        </CustomMockedProvider>
      )
      expect(container).toMatchSnapshot()
    })

    test('with open editable section', async () => {
      const { container, getByTestId } = render(
        <CustomMockedProvider mocks={[mockBuildingBlocks]}>
          <WorkflowDetailBuildingBlocks
            canEdit={true}
            workflow={workflow}
          />
        </CustomMockedProvider>
      )
      fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
      await waitForAllEffectsAndSelectToLoad(container)
      expect(container).toMatchSnapshot()
    })
  })

  test('Should remove a pill', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <WorkflowDetailBuildingBlocks
          canEdit={true}
          workflow={workflow}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)
    fireEvent.click(getByTestId(PILL_REMOVE_BUTTON_TEST_ID))
    expect(screen.queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should add a pill and revert changes on "Cancel" button click', async () => {
    const { container, getByTestId, getByText } = render(
      <CustomMockedProvider mocks={[mockBuildingBlocks]}>
        <WorkflowDetailBuildingBlocks
          canEdit={true}
          workflow={workflow}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)

    fireEvent.keyDown(getByTestId(BUILDING_BLOCK_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)
    fireEvent.click(getByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(2)

    fireEvent.keyDown(getByTestId(BUILDING_BLOCK_SEARCH_TEST_ID).childNodes[1], { key: 'ArrowDown' })
    await screen.findByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)
    fireEvent.click(getByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL))
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(3)

    expect(container).toMatchSnapshot()

    fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
    expect(screen.queryByText(WORKFLOW_TEST_BUILDING_BLOCK_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()

    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    await waitForAllEffectsAndSelectToLoad(container)

    expect(screen.queryByText(WORKFLOW_TEST_BUILDING_BLOCK_LABEL)).toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_1_LABEL)).not.toBeInTheDocument()
    expect(screen.queryByText(BUILDING_BLOCK_SEARCH_OPTION_2_LABEL)).not.toBeInTheDocument()
    expect(screen.queryAllByTestId(PILL_TEST_ID)).toHaveLength(1)
  })
})
