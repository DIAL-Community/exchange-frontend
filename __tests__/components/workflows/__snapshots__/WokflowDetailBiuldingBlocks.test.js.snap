// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the WorkflowDetailBuildingBlocks component. Should add a pill and revert changes on "Cancel" button click 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r2:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-16oyauc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r2:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r2:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Building Block
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Building Block 1
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Building Block 2
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the WorkflowDetailBuildingBlocks component. Should match snapshot - with edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      >
        <button
          class="mr-2 inline-flex items-center gap-x-1.5 bg-dial-blue px-2 py-1 rounded-md text-white"
          data-testid="edit-button"
          type="button"
        >
          <img
            alt="Edit"
            height="12px"
            src="/icons/edit.svg"
            width="12px"
          />
          <span
            class="text-sm"
          >
            Edit
          </span>
        </button>
      </div>
    </div>
    <div
      class="grid grid-cols-1"
    >
      <a
        href="/building_blocks/test_building_block"
      >
        <div
          class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-building-block hover:text-dial-yellow
"
        >
          <div
            class="bg-white border border-dial-gray hover:border-transparent card-drop-shadow"
          >
            <div
              class="grid grid-cols-12 gap-x-4 py-4 px-4"
            >
              <div
                class="col-span-10 pr-3 text-base font-semibold 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Logo for: Test Building Block"
                    class="building-block-filter inline mr-2 "
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="ml-8 w-4/5 h-3/5 relative 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
                >
                  Test Building Block
                </span>
              </div>
              <div
                class="col-span-2 flex flex-row font-semibold opacity-50 justify-end"
              >
                BETA
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the WorkflowDetailBuildingBlocks component. Should match snapshot - with open editable section 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r0:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-16oyauc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r0:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r0:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      >
        <div
          class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-lg text-dial-gray-dark"
          data-testid="pill"
        >
          Test Building Block
          <svg
            class="ml-3 text-2xl cursor-pointer"
            data-testid="remove-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the WorkflowDetailBuildingBlocks component. Should match snapshot - without edit permission. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="grid grid-cols-1"
    >
      <a
        href="/building_blocks/test_building_block"
      >
        <div
          class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-building-block hover:text-dial-yellow
"
        >
          <div
            class="bg-white border border-dial-gray hover:border-transparent card-drop-shadow"
          >
            <div
              class="grid grid-cols-12 gap-x-4 py-4 px-4"
            >
              <div
                class="col-span-10 pr-3 text-base font-semibold 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Logo for: Test Building Block"
                    class="building-block-filter inline mr-2 "
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
                <span
                  class="ml-8 w-4/5 h-3/5 relative 
   whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
                >
                  Test Building Block
                </span>
              </div>
              <div
                class="col-span-2 flex flex-row font-semibold opacity-50 justify-end"
              >
                BETA
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the WorkflowDetailBuildingBlocks component. Should remove a pill 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="mt-12"
    data-testid="editable-section"
  >
    <div
      class="flex justify-between items-center mb-3"
    >
      <div
        class="card-title text-dial-gray-dark"
      >
        Building Blocks
      </div>
      <div
        class="gap-3"
      />
    </div>
    <div
      class="bg-edit p-6"
    >
      <p
        class="card-title text-dial-blue mb-3"
      >
        Assign
         
        Building Blocks
      </p>
      <label
        class="flex flex-col gap-y-2 mb-2"
        data-testid="building-block-search"
      >
        Search and Assign Building Blocks
        <div
          class=" css-l45kjq-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-:r1:-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="react-select__control css-16oyauc-control"
          >
            <div
              class="react-select__value-container css-17t7gkf-ValueContainer"
            >
              <div
                class="react-select__placeholder css-5rjclb-placeholder"
                id="react-select-:r1:-placeholder"
              >
                Type to search...
              </div>
              <div
                class="react-select__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-:r1:-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="react-select__input"
                  id="async-select-id-undefined"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
              >
                <svg
                  class="text-2xl"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        class="flex flex-wrap gap-3 mt-5"
      />
      <div
        class="flex justify-end mt-8 gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
